packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.test.js

```diff
@@ -4,11 +4,46 @@ import { shallow } from 'enzyme';
 import MonthPicker from './MonthPicker.component';
 
 describe('MonthPicker', () => {
-	it('should render a MonthPicker', () => {
-		// when
+	it('should render', () => {
 		const wrapper = shallow(<MonthPicker />);
 
-		// then
 		expect(wrapper.getElement()).toMatchSnapshot();
 	});
+
+	it('should have exactly one selected month', () => {
+		const monthIndex = 8;
+		const wrapper = shallow(<MonthPicker
+			selectedMonthIndex={monthIndex}
+		/>);
+
+		const actions = wrapper.find('PickerAction');
+
+		expect(actions.length).toBe(12);
+
+		const selectedActions = actions.filterWhere(action => action.prop('isSelected') === true);
+
+		expect(selectedActions.length).toBe(1);
+		const selectedAction = selectedActions.first();
+		expect(selectedAction.parent().key()).toBe(monthIndex.toString());
+	});
+
+	it('should callback with the month index picked', () => {
+		const monthIndexToSelect = 5;
+		const onSelect = jest.fn();
+
+		const wrapper = shallow(<MonthPicker
+			onSelect={onSelect}
+		/>);
+
+		const juneAction = wrapper
+			.findWhere(n => n.key() === monthIndexToSelect.toString())
+			.first()
+			.find('PickerAction');
+
+		expect(juneAction.length).toBe(1);
```

Comment:
This is not necessary. Moreover, it will always pass as you get the first with the wanted key.
And the fact that there is only 1 june is tested with the snapshot of the first test.
Just remove it :)

Reply:
✔